Tests/CacheAdvanceTests/CacheAdvanceTests.swift

```diff
@@ -96,20 +96,19 @@ final class CacheAdvanceTests: XCTestCase {
         XCTAssertEqual(messages, [])
     }
 
-    func test_messages_throwsFileCorruptedWhenOffsetInFileAtEndOfNewsetMessageOutOfSync() throws {
-        let randomHighValue: UInt64 = 10_1000
+    func test_messages_throwsFileCorruptedWhenOffsetInFileAtEndOfNewestMessageIsBeyondLastMessageInEmptyFile() throws {
+        let randomHighValue: UInt64 = 101_000
         let header = try CacheHeaderHandle(
             forReadingFrom: testFileLocation,
             maximumBytes: randomHighValue,
             overwritesOldMessages: true)
         let cache = CacheAdvance<TestableMessage>(
             fileURL: testFileLocation,
             writer: try FileHandle(forWritingTo: testFileLocation),
-            reader: try CacheReader(forReadingFrom: testFileLocation),
-            header: try CacheHeaderHandle(
+            reader: try CacheReader(
                 forReadingFrom: testFileLocation,
-                maximumBytes: header.maximumBytes,
-                overwritesOldMessages: header.overwritesOldMessages),
+                maximumBytes: randomHighValue),
+            header: header,
```

Comment:
oops. we were previously re-creating the header rather than using the one we already had.